cs130_back/handlers/user.go

```diff
@@ -489,6 +489,120 @@ func GetClassesInfo(db *gorm.DB, w http.ResponseWriter, r *http.Request) {
 	respondWithJSON(w, http.StatusOK, response)
 }
 
+// UserSearchDetails contains details displayed during user search
+type UserSearchDetails struct {
+	ID				int
+	FirstName 		string		
+	LastName		string
+	Email			string
+
+}
+
+// GetAllUsers retrieves all users in the DB (for search functionality)
+func GetAllUsers(db *gorm.DB, w http.ResponseWriter, r *http.Request) {
+	response := make(map[string][]UserSearchDetails)
+
+	rows, err := db.Model(&models.User{}).Select("id, first_name, last_name, email").Rows()
```

Comment:
Not that important since this is more of a convention thing, but you should ideally create a local User Model variable, have a select function in the user models, and then call it from the user var you created. Just a note, but I think you can keep it as is.

Reply:
Makes sense. I'll change this in a future commit